modules/library/cql/src/main/java/org/geotools/filter/text/commons/ExpressionToText.java

```diff
@@ -168,6 +176,17 @@ public Object visit(Literal expression, Object extraData) {
         Object literal = expression.getValue();
         if (literal instanceof Geometry) {
             Geometry geometry = (Geometry) literal;
+            if (geometry.getUserData() instanceof CoordinateReferenceSystem) {
```

Comment:
@aaime this is the biggest problem that I have with this PR. If I understand this correctly, every geometry that has a CRS set will be encoded with `SRID=epsgCode`. What if GeoTools is being used to encode CQL that is then passed to third-party software that does not support this extension? If this is the case, this behavioural change risks breaking interoperability. Could application code use `ExpressionToText` for this purpose? Do we need a switch to control this behaviour?

Reply:
I don't think many apps do round trip CQL (you cannot even round trip a multipoint geometry for example, it expects a different syntax in read and write) but one is @jnh5y GeoMesa. Jim, do you have an opinion on this one?
Ideas:
* Limit it to ECQL (disable in CQL)
* Guard it with a system variable?

Reply:
Let's see...  in general, I'm for being more clear about what's going on.  To say that a geometry intersects a polygon leaves the context (CRS) unspecified and hence open to interpretation.  

I think either of your suggestions would help.  

For what it is worth, I think Ben's issue is separate from ECQL.toFilter(ECQL.toCQL) and ECQL.toCQL(ECQL.toFilter) each being 'well-behaved'.  (Tersely, the compositions should be the identity function on equivalence classes of CQL strings and GeoTools Filter instances.)

On the plus side, I think that disabling the SRID output in the CQL version of the visitor knocks out Ben's concern.  

For folks who want the 'round-trip' property (or as close as possible), then the ECQL version should work.